### PR TITLE
Add method in Node to resolve a topic or service name

### PR DESCRIPTION
**Pull request by ivanpauno**
Adds bindings for https://github.com/ros2/rcl/pull/835.

Equivalent to https://github.com/ros2/rclcpp/pull/1410.
Will be useful in https://github.com/ros2/rclpy/pull/635.

### CODE REVIEW

**Comment by ivanpauno** [comment]
@ros-pull-request-builder retest this please

File: rclpy/rclpy/node.py [hunk 1]
```diff
@@ -1091,6 +1091,21 @@ def remove_waitable(self, waitable: Waitable) -> None:
         self.__waitables.remove(waitable)
         self._wake_executor()
 
+    def resolve_topic_or_service_name(
```
Comment (jacobperron) [inline rclpy/rclpy/node.py:1094]: At this layer, it seems a little clunky to me to pass a separate flag for a service. Though I don't have particular use-case in mind, I think I would prefer two separate functions:

- `resolve_topic_name()`
- `resolve_service_name()`

**Review by jacobperron — APPROVED** [review]
One comment; take it or leave it.

File: rclpy/rclpy/node.py (hunk 1 above)
Comment (ivanpauno) [reply rclpy/rclpy/node.py]: Comment taken

**Comment by ivanpauno** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12917/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7870/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10633/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12866/)

File: rclpy/rclpy/node.py [hunk 2]
```diff
@@ -1091,6 +1091,31 @@ def remove_waitable(self, waitable: Waitable) -> None:
...
+        with self.handle as capsule:
+            return _rclpy.rclpy_resolve_name(capsule, topic, only_expand, False)
+
+    def resolve_service_name(
+        self, service: str, *, only_expand: bool = False, is_service: bool = False
+    ) -> str:
+        """
+        Return a service name expanded and remapped.
+
+        :param name: service name to be expanded and remapped.
```
Comment (jacobperron) [inline rclpy/rclpy/node.py:1111]:
```suggestion
        :param service: service name to be expanded and remapped.
```

File: rclpy/rclpy/node.py [hunk 3]
```diff
@@ -1091,6 +1091,31 @@ def remove_waitable(self, waitable: Waitable) -> None:
...
+
+        :param only_expand: if `True`, remapping rules won't be applied.
+        :return: a fully qualified topic name,
+            result of applying expansion and remapping to the given `topic`.
+        """
+        with self.handle as capsule:
+            return _rclpy.rclpy_resolve_name(capsule, topic, only_expand, False)
+
+    def resolve_service_name(
+        self, service: str, *, only_expand: bool = False, is_service: bool = False
```
Comment (jacobperron) [inline rclpy/rclpy/node.py:1106]:
```suggestion
        self, service: str, *, only_expand: bool = False
```

File: rclpy/rclpy/node.py [hunk 4]
```diff
@@ -1091,6 +1091,31 @@ def remove_waitable(self, waitable: Waitable) -> None:
         self.__waitables.remove(waitable)
         self._wake_executor()
 
+    def resolve_topic_name(self, topic: str, *, only_expand: bool = False) -> str:
+        """
+        Return a topic name expanded and remapped.
+
+        :param only_expand: if `True`, remapping rules won't be applied.
```
Comment (jacobperron) [inline rclpy/rclpy/node.py:1098]: We should document the `topic` argument.

Comment (ivanpauno) [reply rclpy/rclpy/node.py:1098]:
```suggestion
        :param topic: topic name to be expanded and remapped.
        :param only_expand: if `True`, remapping rules won't be applied.
```

**Comment by ivanpauno** [comment]
I'm not sure why mac failed before, there shouldn't be anything platform dependent here:
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10638/)

File: rclpy/src/rclpy/_rclpy.c [hunk 5]
```diff
@@ -1693,6 +1693,69 @@ rclpy_remap_topic_name(PyObject * Py_UNUSED(self), PyObject * args)
...
+ * \param[in] is_service `true` for service names, `false` for topic names
+ * \return expanded and remapped topic name
+ */
+static PyObject *
+rclpy_resolve_name(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pynode = NULL;
+  PyObject * pyname = NULL;
+  bool only_expand = false;
+  bool is_service = false;
```
Comment (jacobperron) [inline rclpy/src/rclpy/_rclpy.c:1713]: I think using `const char *` for string is correct, but maybe try using `int` for the bool values (just a shot in the dark).

Comment (ivanpauno) [reply rclpy/src/rclpy/_rclpy.c]: :bow: I think that was an accurate short in the dark.
I know "p" is expecting `int`, but I always forget it :grimacing: (and untyped varargs don't help).

**Comment by ivanpauno** [comment]
:crossed_fingers: 
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10646/)

**Comment by ivanpauno** [comment]
Force pushed to fix missing DCO signing.

**Review by jacobperron — APPROVED** [review]

**Pull request merged**